### PR TITLE
chore: update-workflows

### PR DESCRIPTION
**Pull request by misitebao**
- [x] Merge push, test, PR workflows
- [x] Simplify sponsors image generation workflow
- [x] Add auto sync translation document workflow

### CODE REVIEW

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -23,24 +23,64 @@ jobs:
         if: matrix.os == 'ubuntu-latest'
         run: sudo apt-get update -y && sudo apt-get install -y libgtk-3-dev libwebkit2gtk-4.0-dev build-essential pkg-config
 
-      - name: Set up Go 1.18
+      - name: Setup Go
```
Comment (leaanthony) [inline .github/workflows/build-and-test.yml:26]: When I looked at this, the tests did not seem to go into the `v2` directory. I tried using `working-directory: ./v2/` but that generated some build constraint errors that I haven't investigated yet.

File: .github/workflows/pr.yml [hunk 2]
```diff
@@ -1,30 +1,51 @@
 name: PR Checks
+
 on:
   pull_request:
   pull_request_review:
-    types: [ submitted ]
+    types: [submitted]
 
 jobs:
   check_branch_name:
```
Comment (leaanthony) [inline .github/workflows/pr.yml:9]: I think I want to get rid of this check. It's artificial and generates noise. I guess we don't care about source branch name in the end.

File: .github/workflows/sync-translated-documents.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    if: github.repository == 'wailsapp/wails'
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set node
+        uses: actions/setup-node@v2
+        with:
+          node-version: 16.x
+
+      - name: Update sponsors
```
Comment (leaanthony) [inline .github/workflows/sync-translated-documents.yml:18]: Is this the right now?

**Review by leaanthony — COMMENTED** [review]
Just some comments... 👍

File: .github/workflows/pr.yml (hunk 2 above)
Comment (misitebao) [reply .github/workflows/pr.yml:9]: Meaning there is no need to check the branch name of the PR, right?

File: .github/workflows/sync-translated-documents.yml (hunk 3 above)
Comment (misitebao) [reply .github/workflows/sync-translated-documents.yml:18]: Oh, this is my mistake. 😭

File: .github/workflows/build-and-test.yml (hunk 1 above)
Comment (misitebao) [reply .github/workflows/build-and-test.yml:26]: `/wails/v2/cmd/wails/internal/commands/initialise/templates/templates/vue/main.tmpl.go` is a template file, but `//go:embed` in the file will cause an error in the `go test -v ./...` command: 
```
$ go test -v ./...
cmd\wails\internal\commands\initialise\templates\templates\vue\main.tmpl.go:11:12: pattern all:frontend/dist: no matching files found
```

Comment (leaanthony) [reply .github/workflows/build-and-test.yml:26]: We can call it `main.go.tmpl` and it will still work 👍

**Review by leaanthony — APPROVED** [review]

File: .github/workflows/build-and-test.yml (hunk 1 above)
Comment (misitebao) [reply .github/workflows/build-and-test.yml:26]: Will be fixed in this PR https://github.com/wailsapp/wails/pull/2169

**Pull request merged**